x-pack/plugins/rollup/public/crud_app/sections/job_create/steps/components/cron_editor/cron_editor.js

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { padLeft } from 'lodash';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import {
+  EuiSelect,
+  EuiText,
+  EuiFormRow,
+} from '@elastic/eui';
+
+import {
+  getOrdinalValue,
+  getDayName,
+  getMonthName,
+  cronExpressionToParts,
+  cronPartsToExpression,
+  MINUTE,
+  HOUR,
+  DAY,
+  WEEK,
+  MONTH,
+  YEAR,
+} from '../../../../../services';
+
+import { CronHourly } from './cron_hourly';
+import { CronDaily } from './cron_daily';
+import { CronWeekly } from './cron_weekly';
+import { CronMonthly } from './cron_monthly';
+import { CronYearly } from './cron_yearly';
+
+function makeSequence(min, max) {
```

Comment:
this seems somewhat inefficient since the ranges and subsequent values created from this loop are all fixed and will not change over time. I'm not completely sure how to do away with this so I'm adding this comment to open discussion.

could the `MINUTE_OPTIONS` and `HOUR_OPTIONS` be hardcoded constants (in another file)?

could `DAY_OPTIONS` and `MONTH_OPTIONS` be retrieved from moment? then moment-local can take care of localization

`DATE_OPTIONS` is the toughest one. probably can't avoid a `[1...31]` loop here but maybe can be localized with moment as well? examples:
`moment('1/3', 'M/D').format('Do');` returns `3rd`
`moment('1/31', 'M/D').format('Do');` returns `31st`

Reply:
I think you have a good point. I think I'd prefer to just hardcode everything in a constants file and avoid a dependency on moment if we can avoid it.